server/src/main/java/org/elasticsearch/common/settings/ConsistentSettingsService.java

```diff
@@ -44,22 +49,19 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
 
-import javax.crypto.SecretKey;
-import javax.crypto.SecretKeyFactory;
-import javax.crypto.spec.PBEKeySpec;
-
 /**
  * Used to publish secure setting hashes in the cluster state and to validate those hashes against the local values of those same settings.
  * This is colloquially referred to as the secure setting consistency check. It will publish and verify hashes only for the collection
- * of settings passed in the constructor. The settings have to have the {@link Setting.Property#Consistent} property. 
+ * of settings passed in the constructor. The settings have to have the {@link Setting.Property#Consistent} property.
  */
-public final class ConsistentSettingsService {
+public final class ConsistentSettingsService implements ClusterStateListener {
     private static final Logger logger = LogManager.getLogger(ConsistentSettingsService.class);
 
     private final Settings settings;
     private final ClusterService clusterService;
     private final Collection<Setting<?>> secureSettingsCollection;
     private final SecretKeyFactory pbkdf2KeyFactory;
+    private volatile Map<String, Boolean> consistencyCache;
```

Comment:
Do we need this cache, and do we need to populate it on cluster state change ?  I see it the cluster state work is protected by a change in master node ids, but it seems that a simpler solution is check the consistency every time with every call to `areAllConsistent`